### ISSUE TITLE
Why R throw an error "Cannot open Assembly"?

### ISSUE THREAD

**Issue by SumanGajurel** [APSIMInitiative/ApsimX#6815]
Hi team, 
  I have installed ApsimX Next Gen for Mac but it throw an error like this. I have installed mono.framework (version: 6.12.0.140).
<img width="919" alt="Screenshot 2021-09-24 at 4 33 15 pm" src="https://user-images.githubusercontent.com/59353497/134628999-cd7a30fc-e3ea-4f2f-a7da-ee2f91af9ebb.png">

run.string.mac <- "mono /Applications/APSIM2021.9.17.6802.app/Contents/Resources/bin/Models.exe ./Wheat.apsimx"

system(command = run.string.mac, intern = TRUE)

ERROR 
Cannot open assembly '/Applications/APSIM2021.9.17.6802.app/Contents/Resources/bin/Models.exe': No such file or directory.
character(0)
attr(,"status")
[1] 2

Warning message:
In system(command = run.string.mac, intern = TRUE) :
  running command 'mono /Applications/APSIM2021.9.17.6802.app/Contents/Resources/bin/Models.exe ./Wheat.apsimx' had status 2

**Comment by femiguez** [comment]
Recent versions of APSIM Next Gen do not need mono as they are built on the .NET core framework. If you want to run a recent version of APSIM Next Gen (I have tested up to 2021.8.12.6707) you can use the 'apsimx' R package (on CRAN and github). I will be adding the option of running the new framework soon and update the documentation as well.

**Comment by SumanGajurel** [comment]
I did not use the mono version so it is giving me an warning message
<img width="670" alt="Screenshot 2021-09-27 at 7 05 33 am" src="https://user-images.githubusercontent.com/59353497/134824145-457dfc0d-09b6-4707-a2f4-3364cd4d75cb.png">

If I want to use tested version (2021.8.12.6707). Where do I get this??

If I run Optimization with this error it terminates the process in 1 or. seconds.

**Comment by femiguez** [comment]
If you want to run the latest version of APSIM Next Gen and are using Mac you will need to install dotnet (https://dotnet.microsoft.com/download) if you want to run it from the command line (through R) .

After this, I suggest that you install the latest version of apsimx from github. You need the remotes package installed (not necessarily loaded). In the R console,

```
require(remotes)
remotes::install_github("femiguez/apsimx")
library(apsimx)
```
This should allow you to run APSIM through R. If you want a slightly older version of APSIM Next Gen, go to the download page and increase the number of rows displayed. Find APSIM2021.8.12 and download and install that. However, for this version you would first need to install mono if haven't already (https://www.mono-project.com/download/stable/).

In summary, you don't need both mono and dotnet. Mono is for the previous framework 2021.8.12 or older and dotnet is for the current framework.

**Comment by hol430** [comment]
You don't need to install the .net runtime in order to run the latest installed version of apsim on mac - instead, we bundle a self-contained runtime with the installer. If I recall correctly, there is no longer a Models.exe file included in the mac bundle (I think it's called just `Models`). This is probably why the R code is failing.

If you want to install an older version of apsim, you can use the upgrade feature in the gui (click the "show old versions" option). Or you can go to [this page](http://apsimdev.apsim.info/APSIM.Registration.Portal/Main.aspx), which will give you links to all of the released versions of apsim if you've already registered.

**Comment by femiguez** [comment]
@hol430 Thanks for your work on this! I realize that while for the previous versions of APSIM (Aug 2021 and earlier) it was required to install mono just to run APSIM.  This is not needed any more (with .NET Core). My understanding is that the current version ships with the dotnet run time, so this is great for most users and a better long term solution. However, as far as I can tell, in order to be able to run APSIM (current versions) through a script, the 'dotnet' command needs to be installed (is there an alternative?). So this currently works for me:

```
dotnet /Applications/APSIM2021.9.21.6799.app/Contents/Resources/bin/Models.dll Wheat.apsimx
```
But I have the .NET installed, while most users will not. Is there a way to call the binary 'Models'  (without the dll) through the command line which does not require 'dotnet'? I have not been able to do this.

**Comment by SumanGajurel** [comment]
Now it started to work after in install dotnet. Thank you @femiguez and @hol430.

**Comment by hol430** [comment]
Hi @femiguez - you should be able to invoke the Models CLI via `/Applications/APSIM2021.9.21.6799.app/Contents/Resources/bin/Models`. In theory, you should be able to run this without a global .net installation (ie no need to run it with `dotnet` command). I've just tested this and found an issue with our publish command - I'll make a PR to fix this.

Out of interest, would it be more "standard" (or convenient) to have a symlink to this script inside `/Applications/APSIM2021.9.21.6799.app/Contents/MacOS`, alongside the ApsimNG script?

**Comment by femiguez** [comment]
@hol430 Thanks again Drew! From my point of view, since APSIM has used the Models.exe before (7.10) and we have been using bin/Models.exe until recently it might be more intuitive to find it where it is now. It is possible that new users might want to find it alongside ApsimNG, but I would wait until there is an explicit request for this. I plan to document these changes in my 'writing scripts' document so that users can find that as a resource.
